sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -964,11 +970,17 @@ object StaticSQLConf {
     }
   }
 
-  val WAREHOUSE_PATH = buildConf("spark.sql.warehouse.dir")
-    .doc("The default location for managed databases and tables.")
+  val DEFAULT_WAREHOUSE_PATH = buildConf("spark.sql.default.warehouse.dir")
+    .doc("The default location for managed databases and tables. " +
+         "Used if spark.sql.warehouse.dir is not set")
     .stringConf
     .createWithDefault(Utils.resolveURI("spark-warehouse").toString)
 
+  val WAREHOUSE_PATH = buildConf("spark.sql.warehouse.dir")
+    .doc("The location for managed databases and tables.")
```

Comment:
Thats a good point. I misunderstood the meaning of `default` there. Will fix this now